### ISSUE TITLE
Synced tables with a scrollbar have different column's width

### ISSUE THREAD

**Issue by the-goodies** [ocornut/imgui#5920]
Two synced tables with a ImGuiTableFlags_SizingStretchSame and ImGuiTableFlags_ScrollY flags set, have different columns widths if the first table has a vertical scrollbar:

https://user-images.githubusercontent.com/22452788/204001040-5c63a3ec-e329-4aad-aed8-ef9f22b8b9de.mp4

Maybe this is a correct behavior and not a bug, but if I want to align those columns contents, I have to add additional spaces or do some extra layout calculations to position them correctly.
Ideally, in this particular use case there would be a way to freeze the bottom row, so it wouldn't be necessary to create those synced tables.

Code example in Go language:
```go
columnsNum := 6
columnsSetup := func() {
	ui.TableSetupColumn("First")
	ui.TableSetupColumn("Second")
	ui.TableSetupColumn("Third")
	ui.TableSetupColumn("Fourth")
	ui.TableSetupColumn("Fifth")
	ui.TableSetupColumn("And so on")
}

tableID := "##syncedTable"
tableFlags := ui.TableFlagsSizingStretchSame | ui.TableFlagsScrollY | ui.TableFlagsBorders

if ui.BeginTableV(tableID, columnsNum, tableFlags, ui.Vec2{Y: -30}, 0) {
	columnsSetup()
	ui.TableSetupScrollFreeze(0, 1)

	ui.TableHeadersRow()
	for i := 0; i < 60; i += 1 {
		ui.TableNextRow()
		for j := 1; j <= columnsNum; j += 1 {
			ui.TableNextColumn()
			ui.Text(fmt.Sprintf("%d. %d", j, i))
		}
	}

	ui.EndTable()
}

// sum row
if ui.BeginTableV(tableID, columnsNum, tableFlags, ui.Vec2{}, 0) {
	columnsSetup()

	ui.TableNextRow()
	for j := 1; j <= columnsNum; j += 1 {
		ui.TableNextColumn()
		ui.Text(fmt.Sprintf("%d. %d", j, 252352))
	}

	ui.EndTable()
}
```

**Comment by ocornut** [comment]
Thank you for reporting this issue and the repro.

I have now pushed a fix for this 317b33d
![image](https://user-images.githubusercontent.com/8225057/206545364-df7e6a78-dcc0-49a3-94fa-c10eb572e0cf.png)


I have tried 3 approach:

(1)
My first attempt at fixing this was by synchronizing the value of each window DecoOuterX2 to the max of all, which left a hole on the right. 

(2)
Then it seemed simpler instead to synchronize the visibility state of ScrollbarY, meaning other instances would have a full scrollbar, equivalent of using `ImGuiWindowFlags_AlwaysVerticalScrollbar`.

(3)
The obvious possible improvement would be to keep this region usable in instances not needing the scrolling. I've came up with a really simple way to handle that (that's the current commit) but tbh I'm not exactly sure of all the possible side-effects, as synced instanced are currently not thoroughly tested for.

For now it seems to work and passes our existing [test suite](https://github.com/ocornut/imgui_test_engine).
